h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -3489,8 +3862,10 @@ protected Submodel computeSubmodel(int i, double lambda, double nullDevTrain, do
           }
         }
         if (!_parms._HGLM) {  // only perform this when HGLM is not used.
-          if (!_checkPointFirstIter)
+          if (!_checkPointFirstIter) {
             _state.setLambda(lambda);
+            _state.setNoReg();
```

Comment:
Maybe it would be usefull to have a `resetLambda(lambda)` method where the _lambda is set to a new value, and` the _noReg `field will be set to a new value based on reseted` _lambda.`

Reply:
Done.